mathics/builtin/datentime.py

```diff
@@ -1110,12 +1119,23 @@ def eval_3(self, expr, t, failexpr, evaluation):
             try:
                 timeout = float(t.to_python())
                 evaluation.timeout_queue.append((timeout, datetime.now().timestamp()))
-                request = lambda: expr.evaluate(evaluation)
+
+                def run():
+                    return expr.evaluate(evaluation)
+
+                request = run
                 done = False
+                if self.is_running_TimeConstrained:
```

Comment:
A way to check if we have already set a time constraint is to check the length of `evaluation.timeout_queue`. I remember that with the previous implementation, this kind of nested constraint used to work.